transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java

```diff
@@ -183,6 +187,19 @@ public ChannelFuture register() {
         return initAndRegister();
     }
 
+    /**
+     * Create and register a new channel, but asynchronously.
+     * <p>
+     * The returned future completes after the channel has been created and initialised,
+     * but before {@link Channel#register()} has completed.
+     *
+     * @return A future producing the channel object as soon as it has been initialised.
+     */
+    public Future<Channel> registerAsynchronously() {
```

Comment:
Since, `channel.register()` is a rather straightforward method which users can call on the `channel` directly, I think providing a method like `Future<Channel> createUnregistered()` may be more useful such that any other method can be called subsequently on the `Channel`.